docs/rules/require-atomic-updates.md

```diff
@@ -73,22 +91,89 @@ Examples of **correct** code for this rule:
 /* eslint require-atomic-updates: error */
 
 let result;
-async function foo() {
-  result = await somethingElse + result;
 
-  let tmp = await somethingElse;
-  result += tmp;
+async function foobar() {
+    result = await something + result;
+}
+
+async function baz() {
+    const tmp = doSomething(await somethingElse);
+    result += tmp;
+}
+
+async function qux() {
+    if (!result) {
+        const tmp = await initialize();
+        if (!result) {
+            result = tmp;
+        }
+    }
+}
+
+async function quux() {
+    let localVariable = 0;
+    localVariable += await something;
+}
+
+function* generator() {
+    result = (yield) + result;
+}
+```
+
+### Properties
+
+This rule reports an assignment to a property through a variable when it detects the following execution flow in a generator or async function:
+
+1. The variable is read. This can be reading a property through the variable.
```

Comment:
This isn’t clear. Reading a variable from a property? Reading a property and storing it in a variable?

Reply:
This aims to say that `obj` or `obj.prop` can be step 1 for `obj` variable.

```js
obj; // step 1
await something; // step 2
obj.bar = 1; // step 3, error
```

```js
obj.foo; // step 1
await something; // step 2
obj.bar = 1; // step 3, error
```

Reply:
Maybe this then?

```suggestion
1. The variable or object property is read.
```